### DISCUSSION TITLE
new Mac mini issues with get player automator

### DISCUSSION THREAD

**Discussion by dagsuk** [Ascoware/get-iplayer-automator#441] [Q&A]
hi folks. have Mac mini m1 have installed get iplayerautomator on it but every time I stat it it sticks at "updating programme feeds".   and when I try using "use current webpage " I get "status" not available and "show" retrieving meta data        Mac mini is running Mac OS 13.1.    but also have get player automator on Mac book running the same Mac OS   and no issues.    I know it's probably a silly setting I need to change but at the moment I'm stumped.       any help gratefully appreciated. 

thanks in advance. 

Derek

**Comment by dagsuk** [comment]
hi again   change mac mini to mac mini with M2 chip and sadly still having the same problem         :-( 
<img width="793" alt="Screenshot 2023-01-24 at 14 22 07" src="https://user-images.githubusercontent.com/13338035/214319399-b3c238c6-af2e-4a56-93e3-17b809f918e7.png">
<img width="464" alt="Screenshot 2023-01-24 at 14 24 24" src="https://user-images.githubusercontent.com/13338035/214319999-a788bcc7-99d4-4d8e-ae0e-f28113387c8d.png">

**Comment by dagsuk** [comment]
<img width="477" alt="Screenshot 2023-01-24 at 14 27 20" src="https://user-images.githubusercontent.com/13338035/214320750-9efa920d-fcf6-4a2b-b2e1-6169317a50f1.png">

**Comment by skovatch** [comment]
Are you in the UK? Running VPN? Turn on verbose mode to get more details.

**Comment by dagsuk** [comment]
<img width="755" alt="Screenshot 2023-01-24 at 16 54 56" src="https://user-images.githubusercontent.com/13338035/214357083-9c3892cc-f996-42f3-b7d8-37cd74d90a68.png">

**Reply by dagsuk** [reply]
i have it on my MacBook pro and it works fine there  goes straight to "updating programme indexes"and is available within a couple of seconds :-)

**Reply by skovatch** [reply]
The latest version is v1.25.1 but you have 1.24.4.  Are you not able to update?

**Reply by dagsuk** [reply]
how sorry yes have updated version (i was trying a previous version from time machine to see if it made any difference ) 
<img width="484" alt="Screenshot 2023-01-24 at 17 29 18" src="https://user-images.githubusercontent.com/13338035/214365075-c1a239fb-9bd4-4e9a-b534-3585d241ffbd.png">
<img width="590" alt="Screenshot 2023-01-24 at 17 30 29" src="https://user-images.githubusercontent.com/13338035/214365343-52baa236-b432-4137-9aa0-ba014d168087.png">

**Comment by adbeckett** [comment]
I have exactly the same problem - a new M2 Mac Mini running 1.25.1 - I get exactly the same log as above (even in Verbose mode). I have been running it successfully on an M1 MacBook Pro (Ventura) and also an old iMac for many years. I'm in the UK, so no VPN. This was with out-of-the-box settings (I've tried fiddling with various options to no avail). Hard to know what to try next...

**Reply by dagsuk** [reply]
Mines has started working.    Did you do a Time Machine restore ?   I used “cleanmymac” to remove all possible files and folders relating to iPlayer Automator.   Installed and thankfully now all working.  :-)

**Reply by adbeckett** [reply]
It wasn't a restore from Time Machine; because the iMac I'm replacing is old, and an Intel processor, I decided to do a clean install and transfer files across as needed. So the get iplayer automator installation was a completely fresh installation.

**Comment by adbeckett** [comment]
Having turned off all items to cache (which is not a very useful thing of course, but allowed me to check whether downloads were working from a web page), I found that downloading from STV reached 100% but then failed the final step (the same programs succeeded on my MacBook Pro). When it ran a second time (to try to complete the download), the verbose log for Endeavour series 9 episode 1 was as follows:

```
Downloading Show 7/7:

Downloading Endeavour: Series 9
ITV: Requesting Metadata
DEBUG: Metadata response status code: 200
INFO: Metadata processed.
DEBUG: youtube-dl args:["http://player.stv.tv/episode/4euo/endeavour", "--user-agent", "Mozilla/5.0", "-o", "/Volumes/Media Library/TV Shows/Endeavour/Endeavour.s09e01.Sun 26 Feb, 8.00 pm.%(ext)s", "-f", "best[height<=720]", "--verbose", "--embed-thumbnail"]
DEBUG: youtube-dl environment: ["PATH": "/Applications/Get iPlayer Automator.app/Contents/Resources/yt-dlp_macos:/Applications/Get iPlayer Automator.app/Contents/Resources/get_iplayer/utils/bin", "SSL_CERT_FILE": "/Applications/Get iPlayer Automator.app/Contents/Resources/cacert.pem"]
[debug] Command-line config: ['http://player.stv.tv/episode/4euo/endeavour', '--user-agent', 'Mozilla/5.0', '-o', '/Volumes/Media Library/TV Shows/Endeavour/Endeavour.s09e01.Sun 26 Feb, 8.00 pm.%(ext)s', '-f', 'best[height<=720]', '--verbose', '--embed-thumbnail']

[debug] Encodings: locale US-ASCII, fs utf-8, out ascii (No ANSI), err ascii (No ANSI), pref US-ASCII

[debug] yt-dlp version 2021.12.27 [6223f67a8] (mac_dir)

[debug] Python version 3.9.9 (CPython 64bit) - macOS-13.2.1-arm64-64bit

[debug] exe versions: none

[debug] Optional libraries: Cryptodome, sqlite

[debug] Proxy map: {}

[debug] [stv:player] Extracting URL: http://player.stv.tv/episode/4euo/endeavour

[stv:player] 4euo: Downloading webpage

[debug] [brightcove:new] Extracting URL: http://players.brightcove.net/1486976045/default_default/index.html?videoId=6320994730112#__youtubedl_smuggle=%7B%22geo_countries%22%3A+%5B%22GB%22%5D%7D

[debug] Using fake IP 25.198.130.232 (GB) as X-Forwarded-For

[debug] Loading brightcove.1486976045_default from cache

[brightcove:new] 6320994730112: Downloading JSON metadata

[brightcove:new] 6320994730112: Downloading m3u8 information

[brightcove:new] 6320994730112: Downloading m3u8 information

[brightcove:new] 6320994730112: Downloading m3u8 information

[brightcove:new] 6320994730112: Downloading m3u8 information

[brightcove:new] 6320994730112: Downloading MPD manifest

[brightcove:new] 6320994730112: Downloading MPD manifest

[brightcove:new] 6320994730112: Downloading MPD manifest

[brightcove:new] 6320994730112: Downloading MPD manifest

[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id

[info] 6320994730112: Downloading 1 format(s): hls-2132-3

[info] Downloading video thumbnail 8 ...

[info] Writing video thumbnail 8 to: /Volumes/Media Library/TV Shows/Endeavour/Endeavour.s09e01.Sun 26 Feb, 8.00 pm.jpg

[debug] Invoking downloader on "https://manifest.prod.boltdns.net/manifest/v1/hls/v3/clear/1486976045/f135b4b7-9a77-4789-b8a6-c10bcf3f0046/651b955c-2587-4316-97be-e51baa7988a8/99d6dfe4-5a4e-4097-95d3-d2a35e8bb37d/10s/rendition.m3u8?fastly_token=NjQyMGY1NzNfYzNlODBkZmEwYWMxNmVkOTYyMzAyZDM0Mzg0NWFmOGYzMWE1ODM4MmJhMTRjMzQxM2ExY2Y4ZWJiMzRmZDE2Mg%3D%3D"

[download] /Volumes/Media Library/TV Shows/Endeavour/Endeavour.s09e01.Sun 26 Feb, 8.00 pm.mp4 has already been downloaded


[download] 100% of 1.28GiB


WARNING: 6320994730112: Possible MPEG-TS in MP4 container or malformed AAC timestamps

[EmbedThumbnail] Neither mutagen nor AtomicParsley was found. Falling back to ffmpeg

ERROR: Postprocessing: Unable to embed using ffprobe & ffmpeg; ffprobe not found. Please install or provide the path using --ffmpeg-location

Traceback (most recent call last):
  File "yt_dlp/postprocessor/embedthumbnail.py", line 189, in run
  File "yt_dlp/postprocessor/ffmpeg.py", line 255, in get_stream_number
  File "yt_dlp/postprocessor/ffmpeg.py", line 235, in get_metadata_object
yt_dlp.utils.PostProcessingError: ffprobe not found. Please install or provide the path using --ffmpeg-location

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "yt_dlp/YoutubeDL.py", line 3064, in process_info
  File "yt_dlp/YoutubeDL.py", line 3235, in post_process
  File "yt_dlp/YoutubeDL.py", line 3178, in run_pp
  File "yt_dlp/postprocessor/common.py", line 22, in run
  File "yt_dlp/postprocessor/common.py", line 117, in wrapper
  File "yt_dlp/postprocessor/embedthumbnail.py", line 200, in run
yt_dlp.postprocessor.embedthumbnail.EmbedThumbnailPPError: Unable to embed using ffprobe & ffmpeg; ffprobe not found. Please install or provide the path using --ffmpeg-location


yt-dlp finished downloading

```

This is still with 1.25.1 on my new MacMiniM2. I had the same problem with the entire Unforgotten Series 5 and the first episode of Endeavour Series 9 (all from STV, no VPN as I'm the UK).

**Comment by skovatch** [comment]
> [EmbedThumbnail] Neither mutagen nor AtomicParsley was found. Falling back to ffmpeg
> yt_dlp.postprocessor.embedthumbnail.EmbedThumbnailPPError: Unable to embed using ffprobe & ffmpeg; ffprobe not found. Please install or provide the path using --ffmpeg-location

These two lines make me think you have a broken installation, somehow. Can you please open up a Terminal window and run this command:

`ls -al "/Applications/Get iPlayer Automator.app/Contents/Resources/get_iplayer/utils/bin"` 

and paste the output here?

**Reply by adbeckett** [reply]
Hi Scott,

Thanks - the output looks like this:

```
total 151288
drwxr-xr-x@ 4 andrew  staff       128 19 Jan 01:44 .
drwxr-xr-x@ 3 andrew  staff        96 19 Jan 01:43 ..
-rwxr-xr-x@ 1 andrew  staff    309760 19 Jan 01:44 AtomicParsley
-rwxr-xr-x@ 1 andrew  staff  77145664 19 Jan 01:44 ffmpeg

```
This is the same as on my (working) MacBook Pro (M1). I also tried using the following to compute an md5 checksum of all the files in the application and then compute an md5 checksum of the result of that:

```
cd "/Applications/Get iPlayer Automator.app"
find -s . -type f -exec md5 {} \; | sort -k 2 | md5
a171648d9189e943679ae65d4740abcf
```

The result is the same on both computers as well.

Oh, hang on. I see that the "ls -l" output has @ shown after the permissions. So I tried:

`ls -al@ "/Applications/Get iPlayer Automator.app/Contents/Resources/get_iplayer/utils/bin"
`
```
total 151288
drwxr-xr-x@ 4 andrew  staff       128 19 Jan 01:44 .
	com.apple.quarantine	      57 
drwxr-xr-x@ 3 andrew  staff        96 19 Jan 01:43 ..
	com.apple.quarantine	      57 
-rwxr-xr-x@ 1 andrew  staff    309760 19 Jan 01:44 AtomicParsley
	com.apple.quarantine	      57 
-rwxr-xr-x@ 1 andrew  staff  77145664 19 Jan 01:44 ffmpeg
	com.apple.quarantine	      57 
```

I then checked the attributes for the application:

```
xattr /Applications/Get\ iPlayer\ Automator.app
com.apple.macl
com.apple.provenance
com.apple.quarantine
```

I then tried removing the provenance and quarantine attributes (these are not there on my working MacBookPro):

```
xattr -r -d com.apple.quarantine /Applications/Get\ iPlayer\ Automator.app 
xattr -r -d com.apple.provenance /Applications/Get\ iPlayer\ Automator.app
```

That didn't fix it though. It fails in the same way. I'll try deleting the app altogether and re-installing to see if that makes a difference.

**Reply by adbeckett** [reply]
OK, I tried deleting the app and installing it again. If I try to open it (either with double click or ctrl-click and then Open), I get:
![Screenshot 2023-03-03 at 20 36 49](https://user-images.githubusercontent.com/81264259/222825479-86b8e8a5-df0d-4497-aa83-a382c06b2c04.png)
I also tried fixing in the Privacy and Security settings:

![Screenshot 2023-03-03 at 20 38 08](https://user-images.githubusercontent.com/81264259/222825585-e0097079-d474-4e6f-b16f-08940b4940d3.png)
by clicking on the Open Anyway button.

I'd downloaded from [https://github.com/Ascoware/get-iplayer-automator/releases/tag/v1.25.1](https://github.com/Ascoware/get-iplayer-automator/releases/tag/v1.25.1) and then the zip file in the first asset. Is that the right place?

**Reply by adbeckett** [reply]
I meant to say that it still didn't work after doing these steps. I still see the com.apple.quarantine attribute on the files in the directory in question which prevents parts of the app from running, I believe.

**Comment by skovatch** [comment]
Well, I found the root problem... I never built the app with Apple's notarization services, which apparently I should have been doing for the last two years now. I built it with some manual steps, but I need to automate it. 

It sounds like it's the same as the original problem described above. I should have an update within a day or two.

**Reply by adbeckett** [reply]
Thanks Scott. I'd been wondering whether it might be that, and I'd got away with it on my other MacBook Pro as it had been installed pre-Ventura and only updated via the self-update within Get iPlayer Automator; installing from scratch in Ventura 13.2.1 seems to be more fussy about notarization.

I'll test as soon as you have an update - much appreciated.

**Comment by adbeckett** [comment]
Scott,

I just saw an email notifier from you with a test build, but the post isn't here (so perhaps you removed it?). Anyway, I tried it and it had the same problem (maybe that's why you removed it?)

Andrew

**Reply by skovatch** [reply]
The archive was broken so I deleted the link. I'm trying again right now.

**Comment by skovatch** [comment]
OK, one more time: https://drive.google.com/file/d/1-Hk4md-lbbAZM-CIBZYSXf8htF8LS-dn/view?usp=sharing

**Reply by skovatch** [reply]
If the Finder claims the application is damaged, select it in the Finder and duplicate it. Then run the copy.

**Reply by adbeckett** [reply]
Scott,

I tried this download, and I get this on first invocation:

![Screenshot 2023-03-05 at 07 11 49](https://user-images.githubusercontent.com/81264259/222947088-7915ab9b-747b-42fe-a929-3689c1a02fda.png)

After running, it has the same problem (downloads don't complete, same problem as before; caching doesn't work so I have to turn it off) - and the ls -al@ output is just as it was before.

I don't get any indication that anything is damaged.

I wonder (this is wild speculation) whether notarized apps should be distributed as a dmg file rather than zip (just wondering whether zip loses something in the process). I have nothing to back this up - as I said, this is just speculation.

Andrew

**Reply by skovatch** [reply]
Actually, this is an improvement in that notarization has indeed worked. That's a standard warning for apps that aren't distributed via the App Store. Unfortunately it wasn't the root problem.

Oh, I just noticed something: ffmpeg, AtomicParsley, and the lot are still x86_64 only, which means you need Rosetta to run them. In Terminal, do this:

`/usr/sbin/softwareupdate –install-rosetta –agree-to-license`

Then you should be good.

**Reply by adbeckett** [reply]
Scott, that's fixed it. I had to retype the command as it had the wrong kind of hyphens and needs double-hyphen before each argument, but doing that and then installing Get iPlayer Automator again worked perfectly.

I'd not even noticed on my MacBook Pro (M1) that I'd had to install Rosetta - probably because at the time I got that there were still a lot of the apps I was using which were still x86 and so it (I think) prompts you to install it (I'd forgotten), whereas with the MacMini every app I was using was native Apple Silicon.

At least the mystery is now solved! Caching is working fine, and downloads are behaving too.

Thanks for all your help with this; much appreciated.

Andrew

**Reply by Beechside** [reply]
Just to be quite clear, here is the corrected terminal command, corrected as per @adbeckett:

/usr/sbin/softwareupdate --install-rosetta --agree-to-license
